experiments/run-big-bench.py

```diff
@@ -273,10 +270,16 @@ def train_index(
             num_threads=num_build_threads,
         )
 
-        # Now extract the base layer's graph and save it to a file.
-        # This will be a Matrix Market file that we use to construct the Flatnav index.
-        hnsw_index.save_base_layer_graph(filename=hnsw_base_layer_filename)
-
+        try:
+            # Now extract the base layer's graph and save it to a file.
+            # This will be a Matrix Market file that we use to construct the Flatnav index.
+            hnsw_index.save_base_layer_graph(filename=hnsw_base_layer_filename)
+            
+        finally:
+            # delete the HNSW index and force garbage collection
+            del hnsw_index
+            gc.collect()
```

Comment:
This will allow us to save a lot memory especially for billion-scale datasets. For instance, if you run on a machine with 256 threads, HNSW will allocate up to 256 visited lists (one for each thread). Each list takes up around 1GB of memory, so totals up to ~256 GBs just for visited lists. It's good to free up this memory right here.